library_generation/utilities.sh

```diff
@@ -273,3 +273,24 @@ get_version_from_versions_txt() {
   version=$(grep "$key:" "${versions}" | cut -d: -f3) # 3rd field is snapshot
   echo "${version}"
 }
+
+detect_os_architecture() {
+  case "${OSTYPE}" in
+    "linux-gnu"*|"freebsd"*)
+      os_architecture="linux-$(uname -m)"
+      ;;
+    "darwin"*)
+      os_architecture="osx-$(uname -m)"
```

Comment:
When I run `uname -m` on my machine, the output is `arm64`.
However, the right os architecture should be osx-aarch_64.
You can reference the os architecture [here](https://github.com/protocolbuffers/protobuf/releases)

I also implemented a os architecture detection function:
```
get_os_architecture() {
  local os_type
  local os_architecture
  os_type=$(uname -sm)
  case "${os_type}" in
    *"Linux x86_64"*)
      os_architecture="linux-x86_64"
      ;;
    *"Darwin x86_64"*)
      os_architecture="osx-x86_64"
      ;;
    *)
      os_architecture="osx-aarch_64"
      ;;
  esac
  echo "${os_architecture}"
}
```

Reply:
ah, thanks for the function :) I couldn't confirm for the `aarch64` case, but I switched the detection function to this one